### ISSUE TITLE
Warning: .then() only accepts functions but was passed: [object Object]

### ISSUE THREAD

**Issue by markogresak** [jaws-framework/JAWS#126]
I was trying out JAWS by following the quick start guide. When I try to use `jaws dash` to deploy selected resources, I get this warning message:

```
Warning: .then() only accepts functions but was passed: [object Object]
    at CMD.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/deploy_endpoint.js:67:8)
From previous event:
    at Object.module.exports.run (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/deploy_endpoint.js:29:18)
    at CMD.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/dash.js:114:34)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
From previous event:
    at CMD.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/dash.js:107:8)
From previous event:
    at Object.module.exports.run (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/dash.js:28:18)
    at Command.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/bin/jaws:246:23)
    at Command.listener (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/node_modules/commander/index.js:301:8)
    at Command.emit (events.js:110:17)
    at Command.parseArgs (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/node_modules/commander/index.js:610:12)
    at Command.parse (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/node_modules/commander/index.js:458:21)
    at Object.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/bin/jaws:252:11)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3
Warning: .then() only accepts functions but was passed: [object Undefined]
    at CMD.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/deploy_endpoint.js:68:8)
From previous event:
    at Object.module.exports.run (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/deploy_endpoint.js:29:18)
    at CMD.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/dash.js:114:34)
    at processImmediate [as _immediateCallback] (timers.js:367:17)
From previous event:
    at CMD.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/dash.js:107:8)
From previous event:
    at Object.module.exports.run (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/lib/commands/dash.js:28:18)
    at Command.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/bin/jaws:246:23)
    at Command.listener (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/node_modules/commander/index.js:301:8)
    at Command.emit (events.js:110:17)
    at Command.parseArgs (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/node_modules/commander/index.js:610:12)
    at Command.parse (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/node_modules/commander/index.js:458:21)
    at Object.<anonymous> (/usr/local/Cellar/nvm/0.26.1/versions/node/v0.12.7/lib/node_modules/jaws-framework/bin/jaws:252:11)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3
```

I have first tested it in node v4.0.0 and the above is from node v0.12 (both installed via `nvm`), but both result in the same warning message.
The deploy still finishes, but the linked endpoint is unresolvable.

**Comment by ac360** [comment]
@markogresak Sorry!  We're about to merge a huge refactor of JAWS into the master.  This will happen in the next 2 hours.  This issue should be resolved then.

**Comment by markogresak** [comment]
Thanks for the reply, I'll wait for the next version to be available.

**Comment by markogresak** [comment]
Fixed in v1.0.0.
